### DISCUSSION TITLE
mkdocs-material/blog/2021/search-better-faster-smaller/

### DISCUSSION THREAD

**Discussion by giscus** [squidfunk/mkdocs-material#3461] [_]
# mkdocs-material/blog/2021/search-better-faster-smaller/

How we rebuilt client-side search, delivering a better user experience while making it faster and smaller at the same time

https://squidfunk.github.io/mkdocs-material/blog/2021/search-better-faster-smaller/

**Comment by squidfunk** [comment]
Comments are enabled again - we switched to [giscus.app](https://giscus.app), a much better solution than Disqus, as it's ad-free and uses GitHub discussions as a backend.

**Reply by tahamaddam** [reply]
If I apply giscus site-wide, will it provide unique comment section on each page (like in Disqus)?

**Reply by squidfunk** [reply]
Yes.

**Reply by Andre601** [reply]
It primarely depends on what your configuration is for how it should "search" for discussions.
Best option would probably be URL, as that is the most guaranteed to be unique...

**Reply by tahamaddam** [reply]
> It primarely depends on what your configuration is for how it should "search" for discussions.
> Best option would probably be URL, as that is the most guaranteed to be unique...

Would you please explain this? I am using this article as a reference https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/

**Reply by Andre601** [reply]
From https://giscus.app

![image](https://user-images.githubusercontent.com/11576465/153718152-d3cc40d3-01a0-4618-ad97-468410148271.png)

**Comment by hichembd** [comment]
Super

**Reply by hichembd** [reply]
Yess

**Comment by hichembd** [comment]
Good

**Comment by ms2art** [comment]
I'm seeing repeated "Initializing search" messages in the search pop-up as I navigate to different pages in my documentation. Each time search initializes, there's about a 17 second wait before I can enter a search term. I'm using the navigation.instant feature, but it doesn't help. Is this an okay place to ask for help? In case I'm doing something weird in mkdocs.yml, here's what my file looks like, exclusive of the nav section, which lists about 7000 topics. Thanks for any insights.

site_name: 'My Documentation'
docs_dir: source/docs/documentation
site_dir: docs
use_directory_urls: false

theme:
  name: material
  features:
    - navigation.instant

markdown_extensions:
  - md_in_html

plugins:
  - section-index
  - search

**Reply by squidfunk** [reply]
Please [create an new issue](https://github.com/squidfunk/mkdocs-material/issues/new?assignees=&labels=&template=bug.yml) with a minimal reproducible example. I'll look into it then.
